lib/mergeset/table.go

```diff
@@ -198,7 +198,7 @@ func (riss *rawItemsShards) addIbsToFlush(tb *Table, ibsToFlush []*inmemoryBlock
 	}
 	riss.ibsToFlush = append(riss.ibsToFlush, ibsToFlush...)
 	if len(riss.ibsToFlush) >= maxBlocksPerShard*cgroup.AvailableCPUs() {
-		ibsToMerge = ibsToFlush
+		ibsToMerge = riss.ibsToFlush
```

Comment:
Can we have test for this case?

Reply:
Actually, there is a test which reveals this - https://github.com/VictoriaMetrics/VictoriaMetrics/blob/9bad52b68737130b46710c6b860935820e28e8a9/lib/storage/storage_test.go#L817

The problem is that this test generates too small amount of metrics to trigger the new code path. Also, this code  path depends on amount of CPUs on the machine which runs testing.
We can add some make target which would explicitly set `GOMAXPROCS` to some in order to make such tests more predictable.
So to make sure this will be covered by a test we will need to:
- bump amount of series generated in the test mentioned above
- add static `GOMAXPROCS` limit to make tests reproducible

Wdyt?

Reply:
@zekker6 , could you add a separate test into `lib/mergeset` for verifying this case?